src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1536,32 +1536,18 @@ private void SetErrorState(string reason)
             SetState(IndexState.Error, ignoreWriteError: true);
         }
 
-        private void HandleOutOfMemoryException(Exception oome, IndexingStatsScope scope)
+        private void HandleOutOfMemoryException(Exception exception, IndexingStatsScope scope)
         {
             try
             {
-                scope.AddMemoryError(oome);
+                scope.AddMemoryError(exception);
                 Interlocked.Add(ref _lowMemoryPressure, LowMemoryPressure);
                 _lowMemoryFlag.Raise();
 
-                var title = $"Out of memory occurred for '{Name}'";
                 if (_logger.IsInfoEnabled)
-                    _logger.Info(title, oome);
-
-                var message = $"Error message: {oome.Message}";
-                var alert = AlertRaised.Create(
-                    null,
-                    title,
-                    message,
-                    AlertType.OutOfMemoryException,
-                    NotificationSeverity.Error,
-                    key: message,
-                    details: new MessageDetails
-                    {
-                        Message = OutOfMemoryDetails(oome)
-                    });
+                    _logger.Info($"Out of memory occurred for '{Name}'", exception);
 
-                DocumentDatabase.NotificationCenter.Add(alert);
+                DocumentDatabase.NotificationCenter.OutOfMemory.Add(_environment, exception);
```

Comment:
This looks a bit fishy to me. Can you please verify that we do not leak _environment here?